docs/js-api.md

```diff
@@ -0,0 +1,178 @@
+# Using Spectral in code
+
+## Example: Linting an OpenAPI document
+
+Spectral includes a number of ready made rules and functions for OpenAPI v2 and v3 documents.
+
+This example uses the OpenAPI v3 rules to lint a document.
+
+```js
+const { Spectral } = require('@stoplight/spectral');
+const { oas3Functions, rules: oas3Rules } = require('@stoplight/spectral/rulesets/oas3');
+// for YAML
+const { parseWithPointers } = require("@stoplight/yaml");
+
+// Using the anonymous async wrapper for the sake of the example.
+const myOAS = parseWithPointers(`
+responses:
+  '200':
+    description: ''
+    schema:
+      $ref: '#/definitions/error-response'
+`)
+
+// an OASv3 document
+const myOAS = {
+  // ... properties in your document
+  responses: {
+    '200': {
+      description: '',
+      schema: {
+        $ref: '#/definitions/error-response',
+      },
+    },
+  },
+  // ... properties in your document
+};
+
+// create a new instance of spectral with all of the baked in rulesets
+const spectral = new Spectral();
+
+spectral.addFunctions(oas3Functions());
+oas3Rules()
+  .then(rules => spectral.addRules(rules))
+  .then(() => {
+    spectral.addRules({
+      // .. extend with your own custom rules
+    });
+
+    // run!
+    spectral.run(myOAS).then(results => {
+      console.log(JSON.stringify(results, null, 4));
+    });
+  });
+```
+
+You can also [add to these rules](#Creating-a-custom-rule) to create a customized linting style guide for your OpenAPI documents.
+
+The existing OAS rules are opinionated. There might be some rules that you prefer to change. We encourage you to create your rules to fit your use case. We welcome additions to the existing rulesets as well!
+
+## Advanced
+
+### Customization
```

Comment:
This section desevres to be in main readme or at least in a separate file.

Reply:
I have moved the concepts to README but I don't think it makes sense to move advanced concepts to a separate file. It's concerns programmatic use cases and splitting this up into smaller files works against readability IMO. To me at least having simple example and advanced usage in one file is pretty useful.

Reply:
It doesn't concern programmatic usage. It describe basic concepts about rules, rulesets etc. We just have an example of defining a rule using API, but you achieve do the same using CLI

Reply:
I removed the part describing concepts and move to readme already